tensorflow/core/kernels/fused_batch_norm_op.cc

```diff
@@ -62,6 +71,121 @@ struct CastIfNecessary<true, Y, X, T> {
   }
 };
 
+template <typename U, typename T>
+DeviceMemory<U> CastDeviceMemory(Tensor* tensor) {
+  return DeviceMemory<U>::MakeFromByteSize(
+      tensor->template flat<T>().data(),
+      tensor->template flat<T>().size() * sizeof(T));
+}
+
+inline se::port::Status ToExecutorStatus(const Status& s) {
+  return s.ok() ? se::port::Status::OK()
+                : se::port::Status(static_cast<se::port::error::Code>(
+                                       static_cast<int>(s.code())),
+                                   s.error_message());
+}
+
+template <typename>
+struct ToTFDataType;
```

Comment:
We can use `DataTypeToEnum<T>::v()` to get the DataType from a C type, so please remove all of these `ToTFDataType` structs.